src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -102,7 +105,7 @@ public function handle(AddProductToOrderCommand $command)
 
         $this->checkProductInStock($command->getProductId()->getValue(), $command->getCombinationId(), $command->getProductQuantity());
 
-        $cart = $this->createNewCart($order);
+        $cart = $this->createNewOrEditExistingCart($order);
```

Comment:
Maybe you could move this here: `$oldCartRules = $cart->getCartRules();`
Instead of L95, this would seem more logical and I think it will ease the merge with @sowbiba 's PR: https://github.com/PrestaShop/PrestaShop/pull/17927

Reply:
:+1: